engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticatorFactory.scala

```diff
@@ -3,15 +3,18 @@ package pl.touk.nussknacker.ui.security.api
 import akka.http.scaladsl.server.directives.AuthenticationDirective
 import akka.http.scaladsl.server.Route
 import com.typesafe.config.Config
+import sttp.client.{NothingT, SttpBackend}
 
-trait AuthenticatorFactory {
+import scala.concurrent.{ExecutionContext, Future}
+
+trait AuthenticatorFactory[-T] {
```

Comment:
I think it's too generic. You can implement AuthenticatorFactory[String], it will be loaded with ServiceLoader and class cast exception will be thrown. I think it will be easier to just use AuthenticatorContext here... If sb doesn't want to use sttpBackend, they just won't use it...